pca_ica/extract_cells.m

```diff
@@ -0,0 +1,353 @@
+
+    % Reshape filters_out into 2D
+    filters_out = reshape(filters_out,h,w,num_filters);
+
+    % Threshold filters - Deal with multiple boundaries
+    masks = zeros(h,w,num_filters);
+    elim = []; % filters to eliminate due to size requirement
+    for idx_filt = 1:num_filters
+        this_filter = filters_out(:,:,idx_filt);
+        [boundaries, ~] = compute_ic_boundary(this_filter, filter_thresh);
+        this_mask = poly2mask(boundaries{1}(:,1), boundaries{1}(:,2), h, w); 
+        s = regionprops(this_mask,'centroid');
+        if isempty(s)
+            cent_out(:,idx_filt) = [0;0];
+            elim(end+1) = idx_filt;
+        else
+            cent_out(:,idx_filt) = s(1).Centroid;
+        end
+        masks(:,:,idx_filt) = this_mask;
+        filters_out(:,:,idx_filt) = this_filter .* this_mask;
+        if nnz(this_mask) <= 10
+            elim(end+1) = idx_filt; %#ok<AGROW>
+        end
+    end
+
+    % Reshape filters_out back into 1D
+    filters_out = reshape(filters_out,h*w,num_filters);
+    
+    % Eliminate tiny blobs
+    elim = unique(elim);
+    if ~isempty(elim)
+        fprintf('\t \t \t %d cells were eliminated due to minimum size requirement(<10 pixels)\n',length(elim));
+    end
+    filters_out(:,elim) = [];
+    masks(:,:,elim) = [];
+    cent_out(:,elim) = [];
+    idx_out = setdiff(1:num_filters,elim);
+end
+
+function new_assignments = compute_reassignments(masks,cent)
+    
+    num_filters = size(masks,3);
+    
+    % Calculate the matrix of centroid distances
+    NN = repmat(sum(cent.^2,1),num_filters,1);
+    Dist = sqrt(max(NN+NN'-2*cent'*cent,0)); %#ok<MHERM>
+    Dist(Dist<1e-3) = inf; % Set diagonals to infinity
+    Dist = Dist < 15; % Retain only the closeby cells
+    
+    % Calculate overlap between closeby cells
+    J_sim = cell(num_filters,1);
+    for idx_filt = 1:num_filters
+        closeby_cells = find(Dist(idx_filt,:)==1);
+        if ~isempty(closeby_cells)
+            sim = zeros(length(closeby_cells),2);
+            for j = 1:length(closeby_cells)
+                sim(j,:) = [closeby_cells(j),compute_overlap(masks(:,:,idx_filt),masks(:,:,closeby_cells(j)))];
+            end
+            J_sim{idx_filt} = sim;
+        end
+    end
+    
+    % Merge cells that have similarity above a threshold;
+    sim_thresh = 0.8;
+    new_assignments = (1:num_filters)';
+    for idx_filt = 1:num_filters
+        if ~isempty(J_sim{idx_filt}) 
+            for j = 1:size(J_sim{idx_filt},1) % for all cells that overlap with this
+                this_sim = J_sim{idx_filt}(j,2);
+                if this_sim > sim_thresh % similarity is above threshold
+                    that_cell = J_sim{idx_filt}(j,1);
+                    new_assignments(new_assignments==new_assignments(that_cell)) = idx_filt;
+                    J_sim{that_cell} = [];
+                end
+            end
+        end
+    end
+    
+end
+
+function overlap = compute_overlap(mask1, mask2)
```

Comment:
Looks pretty similar to `image_op/compute_mask_overlap.m`. Not possible to reuse or augment that function?